### PR TITLE
refactor(ramda/pickBy): add generic to "ObjPred"

### PR DESCRIPTION
**Pull request by deepfunc**
Add generic to type "ObjPred". It offers key checking and hint for predicate function.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://ramdajs.com/docs/#pickBy>
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ramda ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ramda@latest/))
was missing the following properties:
1. sequence
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d96c9ff809bc5840aba5a980c31a016a75aed7ca
</p>

File: types/ramda/index.d.ts [hunk 1]
```diff
@@ -1402,7 +1402,7 @@ export function pickAll(names: readonly string[]): <T, U>(obj: T) => U;
 /**
  * Returns a partial copy of an object containing only the keys that satisfy the supplied predicate.
  */
-export function pickBy<T, U>(pred: ObjPred, obj: T): U;
+export function pickBy<T, U>(pred: ObjPred<T>, obj: T): U;
 export function pickBy(pred: ObjPred): <T, U>(obj: T) => U;
```
Comment (Nicholaiii) [inline types/ramda/index.d.ts:1406]: Should you not also add the generic to `ObjPred` in the curried form too?

**Review by Nicholaiii — CHANGES_REQUESTED** [review]

File: types/ramda/index.d.ts (hunk 1 above)
Comment (deepfunc) [reply types/ramda/index.d.ts]: 👌 I have fixed it.

**Review by Nicholaiii — APPROVED** [review]

**Comment by deepfunc** [comment]
Ready to merge

**Pull request merged**